src/dataflow/src/logging/materialized.rs

```diff
@@ -176,6 +202,40 @@ pub fn construct<A: Allocate>(
                                     ),
                                 }
                             }
+                            MaterializedEvent::Frontier(name, logical, delta) => {
+                                frontier_session.give((
+                                    row_packer.pack(&[
+                                        Datum::String(&name.to_string()),
+                                        Datum::Int64(worker as i64),
+                                        Datum::Int64(logical as i64),
+                                    ]),
+                                    time_ms,
+                                    delta as isize,
+                                ));
+                            }
+                            MaterializedEvent::KafkaConsumerInfo {
+                                consumer_name,
+                                source_id,
+                                partition_id,
+                                rxmsgs,
+                                rxbytes,
+                                txmsgs,
+                                txbytes,
+                                app_offset,
+                                consumer_lag,
+                            } => {
+                                kafka_consumer_info_session.give((
+                                    (consumer_name, source_id, partition_id),
+                                    time_ms,
+                                    DiffPair::new(
+                                        DiffPair::new(
+                                            DiffPair::new(rxmsgs, rxbytes),
+                                            DiffPair::new(txmsgs, txbytes),
+                                        ),
+                                        DiffPair::new(app_offset, consumer_lag),
+                                    ),
```

Comment:
Hmm, these are all definitely mostly pairs. Would it be nicer if this could be a DiffVec instead of nested pairs though?

I imagine something like
```suggestion
                                    DiffVector::new(vec![
                                        DiffPair::new(rxmsgs, rxbytes),
                                        DiffPair::new(txmsgs, txbytes),
                                        DiffPair::new(app_offset, consumer_lag),
                                    ]),
```

would make the match below a little less overwhelming, too (you'd have to be careful to match names up right, as always, but I don't think that's materially worse than the nested pairs.)